src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -67,6 +86,25 @@ class PexRuntimeEnvironment(Subsystem):
         advanced=True,
     )
 
+    @property
+    def run_packaged_firstparty(self) -> bool:
+        if self.options.is_default("run_packaged_firstparty"):
+            warn_or_error(
+                "2.14.0.dev0",
+                "the option --pex-run-packaged-firstparty defaulting to false",
+                softwrap(
+                    """
+                    In Pants 2.14, by default, running a `pex_binary` will actually package the PEX
+                    and run it, as if you ran `package` followed by executing the built PEX.
+
+                    To fix this deprecation, explictly set `run_packaged_firstparty` in the
+                    `[pex]` section of `pants.toml`. Set it to `false` to use the "old" behavior.
+                    Set it to `true` to use the "new" behavior.
+                    """
```

Comment:
New paragraph, something like this:

> When set to `true`, you can still run the binary as before because you can now `run` on a `python_source` target. The simplest way to do this is to use `./pants run path/to/file.py`, which will find the owning `python_source`. Pants will run the file the same way it used to with `pex_binary` targets.